src/Raven.Server/Documents/Replication/DocumentReplicationTransport.cs

```diff
@@ -56,27 +52,28 @@ public class DocumentReplicationTransport : IDisposable
             }
         }
 
-        public long GetLastEtag()
+        public async Task<long> GetLastEtag()
         {
             using (var writer = new BlittableJsonTextWriter(_context, _websocketStream))
             {
                 _context.Write(writer, new DynamicJsonValue
                 {
-                    ["Raven/GetLastEtag"] = true
+                    [Constants.MessageType] = Constants.Replication.MessageTypes.GetLastEtag
                 });
 
                 writer.Flush();
             }
 
-            var lastEtagMessage = _context.ReadForMemory(_websocketStream, null);
+            var lastEtagMessage = await _context.ReadForMemoryAsync(_websocketStream, null);
 
             long etag;
-            if (!lastEtagMessage.TryGet("Raven/LastSentEtag", out etag))
+            if (!lastEtagMessage.TryGet(Constants.Replication.PropertyNames.LastSentEtag, out etag))
                 throw new InvalidDataException(
-                    "Received invalid last etag message. Failed to get Raven/LastSentEtag property from received result");
+                    $"Received invalid last etag message. Failed to get {Constants.Replication.PropertyNames.LastSentEtag} property from received result");
             return etag;
         }
 
+        //TODO : add here logic so reconnection is attempted couple of times before giving up
```

Comment:
Bad idea, see Release It! 5.5 (Fail Fast).
We have error handling around it anyway. Better to have something that is simple.

Remember, loaded documents are probably memory resident because of the OS